### PR TITLE
Add operations for already saved page

### PR DESCRIPTION
**Pull request by rasendubi**
Problem: it is inconvenient to mark article as read. I need to open
web app, search for article, mark it as read.

Solution: add a new dialog for share action, so now I can share page
into app, press "Read" button.

This adds the following dialog
2017-03-12-224904_499x878_scrot

### CODE REVIEW

File: share.android.js [hunk 1]
```diff
@@ -68,13 +68,44 @@ export default class Share extends Component {
...
         () => ShareExtension.close(),
         () => ShareExtension.close())
+    } else if (action === 'Read' || action === 'Unread') {
+      this.setRead(action === 'Read').then(
+        () => ShareExtension.close(),
+        () => ShareExtension.close())
+    } else if (action === 'Delete') {
+      this.deleteArticle().then(
+        () => ShareExtension.close(),
+        () => ShareExtension.close())
```
Comment (drets) [inline share.android.js:108]: What do you think about switch to `switch`?

Bonus point to abstract: `action.then(() => close(), () => close())` (pseudo)

File: share.android.js [hunk 2]
```diff
@@ -116,13 +141,31 @@ export default class Share extends Component {
...
       <ThemeProvider uiTheme={{}}>
         <View style={styles.container}>
-          {(this.state.server && this.state.token)
-          ? this.renderSaveDialog()
-          : this.renderNoAuth()}
+          {(!this.state.server || !this.state.token)
+           ? this.renderNoAuth()
+           : (this.state.article)
+           ? this.renderSavedDialog()
+           : this.renderSaveDialog()}
```
Comment (drets) [inline share.android.js:168]: It's too fancy for my eyes. Could you move one op to separate func?

**Comment by drets** [comment]
Buttons are not aligned. Tomorrow I'll try to fix that.

File: share.android.js (hunk 2 above)
Comment (rasendubi) [reply share.android.js:168]: It might be better to extract the whole conditional.

Comment (drets) [reply share.android.js:168]: (could be, but in general I'm fine with _one_ `?`)

File: share.android.js (hunk 1 above)
Comment (rasendubi) [reply share.android.js:108]: I wish there were `finally` for promises :)

Comment (drets) [inline share.android.js]: Bluebirdjs has one: http://bluebirdjs.com/docs/api/finally.html Maybe we may wrap a regular promise to use bluebirdjs promise?

File: share.android.js (hunk 2 above)
Comment (rasendubi) [reply share.android.js:168]: It's a matter of separating concerns. Moving only two render calls makes no sense here (how would you call the new function?), while moving all of them (and going for `if`) would make a much better function with clear responsibility (choosing dialog version).

Comment (drets) [reply share.android.js:168]: I don't care about separating concerns in this case as you do: I think more about readability.

Comment (drets) [reply share.android.js:168]: After re-reading the function names and trying to come up with the function name which we can use for `(this.state.article) ? this.renderSavedDialog() : this.renderSaveDialog()` I realized that I don't like that we have 2 functions with the really similar names: saved dialog and save dialog. However, I need to run to the train, so can't help you with naming.

Comment (rasendubi) [reply share.android.js:168]: That's all about readability. Compare this
```js
renderAuthed() {
  return this.state.article ?
    this.renderSavedDialog() :
    this.renderSaveDialog()
}

render() {
  return (
    <ThemeProvider uiTheme={{}}>
      <View style={styles.container}>
        {(!this.state.server || !this.state.token)
        ? this.renderNoAuth()
        : this.renderAuthed()}
      </View>
    </ThemeProvider>
  )
}
```
to this:
```js
renderDialog() {
  if (!this.state.server || !this.state.token) {
    return this.renderNoAuth()
  } else if (this.state.article) {
    return this.renderSavedDialog()
  } else {
    return this.renderSaveDialog()
  }
}

render() {
  return (
    <ThemeProvider uiTheme={{}}>
      <View style={styles.container}>
        {this.renderDialog()}
      </View>
    </ThemeProvider>
  )
}
```

**Comment by rasendubi** [comment]
Not sure what bluebird is. Do we already use it?

**Comment by drets** [comment]
selection_121

:crying_cat_face:

**Comment by drets** [comment]
>Buttons are not aligned. Tomorrow I'll try to fix that.

I spent quite a bit of time on it and it seems it's pretty hard to do using material ui dialog.

**Pull request merged**